File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.ArrayDeque;
class Solution {
    boolean solution(String s) {
        ArrayDeque<Character> stack = new ArrayDeque<>();
        char[] ch = s.toCharArray();
        
        for (char a : ch){
            if(a == '('){
                stack.push(a);
            } else {
                if(stack.isEmpty()){
                    return false;
                }
                stack.pop();
            }
        }
        return stack.isEmpty();
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in s: # s 길이만큼 for문 돌기
        if i == '(':# i가 ( 이면
            stack.append(i) # 스택에 push하기
        else: # i가 )이면 pop (단, empty가 아니어야함)
            if not stack: # empty이면 false
                return False
            stack.pop() # 스택에서 pop하기
    if stack: # 스택이 empty가 아니면 false
        return False
    return True